app/screens/convert_gm_to_channel/convert_gm_to_channel_form/convert_gm_to_channel_form.tsx

```diff
@@ -0,0 +1,152 @@
+import {preventDoubleTap} from '@app/utils/tap';
+import Button from '@components/button';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {displayUsername} from '@utils/user';
+
+import {ChannelNameInput} from '../channel_name_input';
+import MessageBox from '../message_box/message_box';
+import {TeamSelector} from '../team_selector';
+
+import {NoCommonTeamForm} from './no_common_teams_form';
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            paddingVertical: 24,
+            paddingHorizontal: 20,
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 24,
+        },
+        errorMessage: {
+            color: theme.dndIndicator,
+        },
+    };
+});
+
+type Props = {
+    channelId: string;
+    commonTeams: Team[];
+    profiles: UserProfile[];
+    locale?: string;
+    teammateNameDisplay?: string;
+}
+
+export const ConvertGMToChannelForm = ({
+    channelId,
+    commonTeams,
+    profiles,
+    locale,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+
+    const [selectedTeam, setSelectedTeam] = useState<Team>();
+    const [newChannelName, setNewChannelName] = useState<string>('');
+    const [errorMessage, setErrorMessage] = useState<string>('Something went wrong. Failed to convert Group Message to Private Channel');
+
+    const submitButtonEnabled = selectedTeam && newChannelName.trim();
+
+    const {formatMessage} = useIntl();
+    const confirmButtonText = formatMessage({
+        id: 'channel_info.convert_gm_to_channel.button_text',
+        defaultMessage: 'Convert to Private Channel',
+    });
+
+    useEffect(() => {
+        if (commonTeams.length > 0) {
+            setSelectedTeam(commonTeams[0]);
+        }
+    }, []);
+
+    const handleOnPress = useCallback(preventDoubleTap(async () => {
+        if (!submitButtonEnabled) {
+            return;
+        }
+
+        const {updatedChannel, error} = await convertGroupMessageToPrivateChannel(serverUrl, channelId, selectedTeam.id, newChannelName);
+        if (error) {
+            if (isErrorWithMessage(error)) {
+                setErrorMessage(error.message);
+            } else {
+                setErrorMessage(formatMessage({id: 'channel_info.convert_gm_to_channel.conversion_error', defaultMessage: 'Something went wrong. Failed to convert Group Message to Private Channel.'}));
+            }
+
+            return;
+        }
+
+        if (!updatedChannel) {
+            setErrorMessage(formatMessage({id: 'channel_info.convert_gm_to_channel.conversion_error', defaultMessage: 'Something went wrong. Failed to convert Group Message to Private Channel.'}));
+            return;
+        }
```

Comment:
Wonder if in any of these two scenarios we want to give more information to the user about what happened. And if not, wonder if we want to leave some trace in the logs for troubleshooting (in case there is no trace from the underlying functions).

Reply:
In the first else, we don't have any message in API, so we do't really know what happened. In the second if, there was no error, but also no updated channel, so no idea what happened. So can't show anything else. Best we can do is to ask to try again, or to contact admin, but thats not a pattern we do in other places AFAIK.

We're logging the error in the function call above, so its always logged. I've now added log for the `if (!updatedChannel)` condition as well now.